source/dofs/dof_renumbering.cc

```diff
@@ -1863,7 +1857,20 @@ namespace DoFRenumbering
     for (unsigned int i=0; i<n_dofs; ++i)
       new_indices[i] = i;
 
-    std::random_shuffle (new_indices.begin(), new_indices.end());
+    // shuffle the elements; the following is essentially the
+    // std::random_shuffle algorithm but uses a predictable
+    // random number generator
+    ::boost::mt19937 random_number_generator;
+    for (unsigned int i=1; i<n_dofs; ++i)
```

Comment:
I took the implementation in the GCC library as a reference, which reads like this:
```
   random_shuffle(_RandomAccessIterator __first, _RandomAccessIterator __last)
    {
      // concept requirements
      __glibcxx_function_requires(_Mutable_RandomAccessIteratorConcept<
            _RandomAccessIterator>)
      __glibcxx_requires_valid_range(__first, __last);

      if (__first != __last)
        for (_RandomAccessIterator __i = __first + 1; __i != __last; ++__i)
          {
            _RandomAccessIterator __j = __first
                                        + std::rand() % ((__i - __first) + 1);
            if (__i != __j)
              std::iter_swap(__i, __j);
          }
    }
```
I think that is functionally equivalent to the one implemented.

I don't care either way. I can change it if you'd like me to, or I can leave it.